### ISSUE TITLE
Multiple errors

### ISSUE THREAD

**Issue by jamesdbrock** [purescript-contrib/purescript-parsing#222]
Should we have multiple errors like in Megaparsec? https://hackage.haskell.org/package/megaparsec/docs/Text-Megaparsec-Error.html#t:ParseErrorBundle

**Comment by Violet-Codes** [comment]
> I'm all for better error messages. However, don't better errors typically mean slower parsing?

Not at all! `ParseContext` just wraps all errors in a section which is an incredibly cheep action,
`bundleErr` just raises the error if its present and otherwise succeeds
and `recoverWith` will only fire the recovery parser on a failure, and as for the recovery parser, these things are usually incredibly simple, aggressively so.

This would not cause any noticeable performance hit. :+1:

**Comment by JordanMartinez** [comment]
No offense but I'd be more convinced by a benchmark than a claim like above.

**Comment by jamesdbrock** [comment]
Also if we're going to do multiple errors then maybe we should do lazy error messages at the same time see #158
